engine/classes/Elgg/Database/UsersTable.php

```diff
@@ -460,7 +460,7 @@ function generateInviteCode($username) {
 	 */
 	function validateInviteCode($username, $code) {
 		// validate the format of the token created by ->generateInviteCode()
-		if (!preg_match('~^(\d+)\.(\w+)$~', $code, $m)) {
+		if (!preg_match('~^(\d+)\.([\w-]+)$~', $code, $m)) {
```

Comment:
To not rely on locale, let's use `[a-zA-Z0-9\-_]` (we also need `_`).

Also this PR needs to go on the 1.11 branch instead.